### PR TITLE
Use BuildKit for cache

### PR DESCRIPTION
**Pull request by arukuka**
# Summary

Use docker buildkit to improve your local development experience

# Purpose

- when developing locally, I would like to reduce the loading time of the large-capacity cache included in the context
- for a comparison of execution times with and without buildkit, please see the results below.

# Contents

- use `mount=type=cache` for local cache
- specify the image name explicitly
- temporarily changed `small_cc` image because buildkit's cache and CI are not enough compatible
  - I will fix by #36 

# Notation

- test small cc test takes long time
  - run tests on locally, or check CI status
    - because buildkit's cache and CI are not enough compatible
    - I will fix by #36 
- following directories are not related when building image
  - directories:
    - `src/ComputationContainer/.cache`
    - `src/ManageContainer/vendor`
  - these are used for checking local build or small test only
- following directories might be load as context sometimes
  - directories:
    - `src/ComputationContainer/.cache`: ~10GB
    - `src/ManageContainer/vendor`: ~20MB
  - When you found this issue, I would appreciate it if you could report and/or fix it.

# Testing Methods Performed

- conditions:
  - machine: AWS EC2 `c6i.4xlarge`
  - storage: EBS gp2

## docker image `dev_cc1` build time

- command: `(time docker-compose build dev_cc1 --progress plain 2>&1) 2>&1 | tee 20220907.log`
- conditions:
  - cache was already made.
  - file which is not code was changed to prevent that layers cache work.

||time|log|
|-|-|-|
|current|5m8.637s|[2022090700.before.cache.log](https://github.com/acompany-develop/QuickMPC/files/9501903/2022090700.before.cache.log)|
|proposal|**0m20.950s**|[2022090701.after.cache.log](https://github.com/acompany-develop/QuickMPC/files/9501737/2022090701.after.cache.log)|

## source code build: `bazel build //:all` time

- command:
  - current: `bazel build //:all` in container
  - proposal: `(time docker-compose build dev_cc1 --progress plain 2>&1) 2>&1 | tee 20220907.log`
    - include time it took to copy execution binaries.
- conditions:
  - initial cache is made.
  - add spaces to `src/ComputationContainer/ComputationParty/ComputationParty.cpp`

||time|log|
|-|-|-|
|current|**23.972s**|[2022090702.before.cache.log](https://github.com/acompany-develop/QuickMPC/files/9501809/2022090702.before.cache.log)|
|proposal|34.2s|[2022090703.after.cache.log](https://github.com/acompany-develop/QuickMPC/files/9501784/2022090703.after.cache.log)|

- this pr is slower than current way

## others

- I merged this PR to working branch for checking that it works
  - If you don't mind, I would be happy to check as well that it works as well

### CODE REVIEW

**Review by mdonaka — APPROVED** [review]
I have confirmed the operation on my local intel mac.
I was surprised at how fast it was. Unbelievable.
Thanks for the update!

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+**/vendor
+**/.cache
+!!!!!!!aaaa
```
Comment (mdonaka) [inline .dockerignore:3]: @arukuka 
What's this line?

**Review by mdonaka — COMMENTED** [review]
Could you please remove the `Test/DbContainer/*`?

File: .dockerignore (hunk 1 above)
Comment (mdonaka) [reply .dockerignore:3]: > !!!!!!!aaaa

Comment (arukuka) [reply .dockerignore:3]: @mdonaka so sorry... I will remove it.

**Comment by arukuka** [comment]
It seems that I made a mistake when rebasing, so I will make it a draft. :bow:

**Comment by arukuka** [comment]
I checked change lists, and remove dir `Test/DbContainer`.

**Review by mdonaka — APPROVED** [review]
I was able to confirm the operation once again.
Thanks for the great changes!

**Comment by arukuka** [comment]
@mdonaka  Thank you for your check!  
I will merge it.

**Pull request merged**